src/components/NavigationBlocker/NavigationBlocker.tsx

```diff
@@ -0,0 +1,56 @@
+import { useContext, useEffect } from 'react';
+import { useBlocker, useNavigate } from 'react-router-dom';
+import { useFormState } from 'react-final-form';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+import routes from 'constants/routes';
+import WizardFormContext from 'contexts/WizardFormContext';
+
+export default function NavigationBlocker() {
```

Comment:
Suggest better name, please. Now role of this component isn't only blocking transitions but also navigation to the next screen